elyra/pipeline/processor_local.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright 2018-2020 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import papermill
+import tempfile
+import time
+
+from datetime import datetime
+from elyra.pipeline import PipelineProcessor, PipelineProcessorResponse
+
+
+class LocalPipelineProcessor(PipelineProcessor):
+    """
+    Local pipeline processor runs a pipeline locally. The scope of this runtime is
+    simply to automate the execution of multiple notebooks from a pipeline as if they
+    were being executed manually. Any additional support is out of the scope of this
+    processor. If a notebook doesn't run using `run all` it will fail in the
+    same way when processed by this processor. Also, any relationship or specific capabilities
+    associated with a particular runtime is not supported by local mode and any additional properties
+    other then the specific file to be executed is ignored by the local processor.
+
+    Note: Execution happens in-place and a ledger of runs will be available at $TMPFILE/elyra/pipeline-name-<timestamp>
+    """
+    _type = 'local'
+
+    def __init__(self, root_dir=None):
```

Comment:
Is `root_dir` really optional?  If it's `None`, `get_absolute_path()` will fail.  So we should remove `=None`

Reply:
I mostly needed this for testing purposes... but if it's not optional, how does it get injected on the kfp runtime?

Reply:
Anyway, removed this